src/policy/protect_payload.rs

```diff
@@ -99,10 +105,25 @@ impl PolicyModule for ProtectPayloadPolicy {
 
         // Step 2: Save sensitives privileged registers
         self.confidential_values_payload = get_confidential_values(ctx);
-        self.write_confidential_registers(ctx, false)
+        self.write_confidential_registers(ctx, false);
+
+        // Step 3: Lock memory
+        mctx.pmp.set(
```

Comment:
Don't you have a new `set_policy` now? :p

Reply:
It wasn't rebased at that time :-)